src/Tests/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsArtifactPostProcessing.cs

```diff
@@ -0,0 +1,249 @@
+
+                var fileContent = new List<string>();
+                using var streamReader = new StreamReader(mergedFile);
+                while (!streamReader.EndOfStream)
+                {
+                    string line = streamReader.ReadLine();
+                    Assert.StartsWith("SessionEnded_Handler_", line);
+                    fileContent.Add(line);
+                }
+
+                Assert.Equal(3, fileContent.Distinct().Count());
+            }
+            else
+            {
+                Assert.Equal(string.Empty, output[output.Count - 5].Trim());
+                Assert.Equal("Attachments:", output[output.Count - 4].Trim());
+
+                int currentLine = 0;
+                for (int i = 3; i > 0; i--)
+                {
+                    currentLine = output.Count - i;
+                    string file = output[currentLine].Trim();
+                    var fileContent = new List<string>();
+                    using var streamReader = new StreamReader(file);
+                    while (!streamReader.EndOfStream)
+                    {
+                        string line = streamReader.ReadLine();
+                        Assert.StartsWith("SessionEnded_Handler_", line);
+                        fileContent.Add(line);
+                    }
+
+                    Assert.True(fileContent.Distinct().Count() == 1);
+                }
+
+                Assert.Equal(output.Count, currentLine + 1);
+            }
+        }
+
+        private void BuildDataCollector()
+            => LazyInitializer.EnsureInitialized(ref s_dataCollectorDll, ref s_dataCollectorInitLock, () =>
+                {
+                    TestAsset testInstance = _testAssetsManager.CopyTestAsset("VSTestDataCollectorSample").WithSource();
+
+                    string testProjectDirectory = testInstance.Path;
+
+                    new BuildCommand(testInstance)
+                        .Execute("/p:Configuration=Release")
+                        .Should()
+                        .Pass();
+
+                    return Directory.GetFiles(testProjectDirectory, "AttachmentProcessorDataCollector.dll", SearchOption.AllDirectories).Single(x => x.Contains("bin"));
+                });
+
+        private void BuildDataCollectorNoMerge()
+            => LazyInitializer.EnsureInitialized(ref s_dataCollectorNoMergeDll, ref s_dataCollectorInitLock, () =>
+            {
+                TestAsset testInstance = _testAssetsManager.CopyTestAsset("VSTestDataCollectorSampleNoMerge").WithSource();
+
+                string testProjectDirectory = testInstance.Path;
+
+                new BuildCommand(testInstance)
+                            .Execute("/p:Configuration=Release")
+                            .Should()
+                            .Pass();
+
+                return Directory.GetFiles(testProjectDirectory, "AttachmentProcessorDataCollector.dll", SearchOption.AllDirectories).Single(x => x.Contains("bin"));
+            });
```

Comment:
This looks so weird. Why do we need static members that holds something that is changed for each test?